addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -425,6 +490,28 @@ def _l10n_in_edi_generate_invoice_json(self, invoice):
                 json_payload["ExpDtls"].update({
                     "Port": invoice.l10n_in_shipping_port_code_id.code
                 })
+        if not invoice._l10n_in_edi_is_managing_invoice_negative_lines_allowed():
+            return json_payload
+
+        # Set nagative lines agains positive lines as discount with same HSN code
+        discount_lines = []
+        # reverse lines to remove from last
+        for discount_line in json_payload['ItemList'][::-1]:
+            if discount_line['AssAmt'] < 0:
+                discount_lines.append(discount_line)
+                json_payload['ItemList'].remove(discount_line)
+        if not discount_lines:
+            return json_payload
+
+        lines_grouped_and_sorted = defaultdict(list)
+        for line in sorted(json_payload['ItemList'], key=lambda i: i['AssAmt'], reverse=True):
+            lines_grouped_and_sorted[discount_group_key(line)].append(line)
+
+        for discount_line in discount_lines:
+            apply_discount_on_lines = lines_grouped_and_sorted.get(discount_group_key(discount_line), [])
+            for apply_discount_on in apply_discount_on_lines:
+                if put_discount_on(discount_line, apply_discount_on):
```

Comment:
we are sure that discount_line != apply_discount_on (was that the reason for the reverse?)

Reply:
yes, because when we add discount lines at the same time we remove them from ItemList so both are the same is never possible

Reply:
Are you sure?  Because users could put a wrong configuration where the discount line is bigger than the normal lines and then it would do weird things?  (should not we give an error in such case?)

Reply:
then error come from government API

Reply:
yes, I am sure